File: Lessons/7 StacksandQues/StoneWall/stoneWall.js
function solution(H) {
  let outputSum = 0;
  let y = 0;
  const arr = [];

  for (let x = 0; x < H.length; x += 1) {
    while (y > 0 && arr[y - 1] > H[x]) {
      y -= 1;
    }
    if (y > 0 && arr[y - 1] === H[x]) {
      continue;
    } else {
      outputSum += 1;
      arr[y] = H[x];
      y += 1;
    }
  }
  return outputSum;
}

File: Lessons/7 StacksandQues/StoneWall/stoneWall.cpp
#include <vector>
int solution(vector<int> &H) {
  int outputSum = 0;
  vector<int> v(H.size());
  int y = 0;
  for (unsigned int x = 0; x < H.size(); x += 1) {
    while (y > 0 && v.at(y - 1) > H[x]) {
      y -= 1;
    }
    if (y > 0 && v[y - 1] == H[x]) {
      continue;
    } else {
      outputSum += 1;
      v[y] = H[x];
      y += 1;
    }
  }
  return outputSum;
}